src/openvino.cc

```diff
@@ -729,19 +733,8 @@ ModelInstanceState::ProcessRequests(
         RequestsRespondWithError(requests, request_count, err);
         return;
       }
-      // FIXME: The OV will segfault when the batch size is not equal to the max
-      // batch size.
       if (total_batch_size != (size_t)max_batch_size) {
-        RequestsRespondWithError(
```

Comment:
We should have a parameter to control when "batch padding" is enabled. I would set the default to be disabled and when disabled we still want to have this error.

Reply:
I understand why there is a case for throwing an error here by default. But I believe it will be better for the backend to use the WAR to run the inference by default for a good user experience. In the verbose mode, we do log the advice to use bs == mbs for a better performance. I can document this behavior in the README.

Reply:
The tradeoff is between user silently getting bad performance or user getting a failure with clear indication of how to fix (either control batch sizes sent to triton or enable batch padding). Usually it is better to *not* have a silent bad behavior. I do agree there should be a section in the docs explaining the options and the tradeoffs.